program/src/instruction.rs

```diff
@@ -1,184 +1,420 @@
-    /// 6. `[]` SPL Token program
-    /// 7.. Extra required accounts for transfer hook
-    ///
-    /// Instruction data: amount of tokens to move
+    /// 
+    /// Instruction data: amount of tokens to move.
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Stake account"
+    )]
+    #[account(
+        2,
+        writable,
+        name = "vault",
+        desc = "Vault token account"
+    )]
+    #[account(
+        3,
+        writable,
+        name = "destination_token",
+        desc = "Destination token account"
+    )]
+    #[account(
+        4,
+        signer,
+        name = "stake_authority",
+        desc = "Stake authority"
+    )]
+    #[account(
+        5,
+        name = "vault_authority",
+        desc = "Vault authority"
+    )]
+    
+    #[account(
+        6,
+        name = "spl_token_program",
+        desc = "SPL Token program"
+    )]
     WithdrawInactiveStake(u64),
 
     /// Harvests holder SOL rewards earned by the given stake account.
-    ///
+    /// 
     /// NOTE: This mostly replicates the logic in the rewards program. Since the
     /// staked tokens are all held by this program, stakers need a way to access
     /// their portion of holder rewards.
     ///
     /// This instruction requires that `unclaimed_rewards` be equal to `0` in
     /// the token vault account. For ease of use, be sure to call the
     /// `HarvestRewards` on the vault account before this.
-    ///
-    /// 0. `[]` Config account
-    /// 1. `[w]` Stake account
-    /// 2. `[w]` Vault token account
-    /// 3. `[]` Holder rewards account for vault token account
-    /// 4. `[w]` Destination account for withdrawn lamports
-    /// 5. `[s]` Stake authority
-    /// 6. `[]` Vault authority, PDA with seeds `['token-owner', stake_config]`
-    /// 7. `[]` Stake token mint, to get total supply
-    /// 8. `[]` SPL Token program
+    #[account(
+        0,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Stake account"
+    )]
+    #[account(
+        2,
+        writable,
+        name = "vault",
+        desc = "Vault token account"
+    )]
+    #[account(
+        3,
+        name = "holder_rewards",
+        desc = "Holder rewards account for vault token account"
+    )]
+    #[account(
+        4,
+        writable,
+        name = "destination", 
+        desc = "Destination account for withdrawn lamports"
+    )]
+    #[account(
+        5,
+        signer,
+        name = "stake_authority",
+        desc = "Stake authority"
+    )]
+    #[account(
+        6,
+        name = "vault_authority",
+        desc = "Vault authority"
+    )]
+    #[account(
+        7,
+        name = "mint",
+        desc="Stake token mint"
```

Comment:
micro-nit: will rustfmt fix the missing spaces?

Reply:
Not, we are disabling `fmt` on this enum so we have a predictable structure for this tags. I will fix the missing spaces.